### DISCUSSION TITLE
Meson version build error

### DISCUSSION THREAD

**Discussion by chlee75** [fwupd/fwupd#4358] [Q&A] [answered]
How to fix fwupd build error due to meson version?

[infineon@fedora build]$ meson ../fwupd
The Meson build system
Version: 0.59.4
Source dir: /home/infineon/Works/LVFS/HP/HOOK2/fwupd
Build dir: /home/infineon/Works/LVFS/HP/HOOK2/build
Build type: native build

../fwupd/meson.build:1:0: ERROR: Meson version is 0.59.4 but project requires >=0.60.0

A full log can be found at /home/infineon/Works/LVFS/HP/HOOK2/build/meson-logs/meson-log.txt

**Comment by hughsie** [comment, marked as answer]
Hey Ryan. You can either do `sudo dnf update --releasever 36 meson` and continue to use the `main` branch or you can switch to the old-meson-using branch by doing `git checkout 1_7_X` -- I hope that helps.

**Comment by chlee75** [comment]
Hi. Richard.

Thanks for your answer.
I have already fixed this issue by upgrading Fedora 35 to Fedora 36. 
[https://www.linuxcapable.com/how-to-upgrade-fedora-35-to-fedora-36-with-gnome-42/](url)

Sorry I did not applied your suggestion to my Fedora 35 because of Fedora kernel upgrade.  
However your answer would be more good solution than upgrading Fedora.

Thanks.

**Comment by chlee75** [comment]
Hi. Richard.

@hughsie 

Is it right "sudo dnf update --releasever=36 meson" instead of "sudo dnf update --releasever 36 meson"?

It seems that Fedora 36 has not officially released yet. 
Is it OK for me to develop fwupd plugin under Fedora 36?

Thanks.
Ryan.

**Reply by hughsie** [reply]
I think both commands should be equivalent. F36 is still in the prerelease phase, but it's quite safe to upgrade meson on F35. It's also probably safe to upgrade everything to F36 too; it's what I'm using now :)
